packages/@apphosting/adapter-angular/src/utils.ts

```diff
@@ -161,39 +166,52 @@ function extractManifestOutput(output: string): string {
  */
 export async function generateOutputDirectory(
   cwd: string,
-  outputPathOptions: OutputPathOptions,
+  outputBundleOptions: OutputBundleOptions,
 ): Promise<void> {
-  await move(outputPathOptions.baseDirectory, outputPathOptions.outputBaseDirectory, {
+  await move(outputBundleOptions.baseDirectory, outputBundleOptions.outputBaseDirectory, {
     overwrite: true,
   });
-  if (outputPathOptions.needsServerGenerated) {
-    await generateServer(outputPathOptions);
+  if (outputBundleOptions.needsServerGenerated) {
+    await generateServer(outputBundleOptions);
   }
-  await generateBundleYaml(outputPathOptions, cwd);
+  await generateBundleYaml(outputBundleOptions, cwd);
 }
 
 // Generate bundle.yaml
 async function generateBundleYaml(
-  outputPathOptions: OutputPathOptions,
+  outputBundleOptions: OutputBundleOptions,
   cwd: string,
 ): Promise<void> {
   await writeFile(
-    outputPathOptions.bundleYamlPath,
+    outputBundleOptions.bundleYamlPath,
     yamlStringify({
       headers: [],
       redirects: [],
       rewrites: [],
-      runCommand: `node ${normalize(relative(cwd, outputPathOptions.serverFilePath))}`,
-      neededDirs: [normalize(relative(cwd, outputPathOptions.outputDirectory))],
-      staticAssets: [normalize(relative(cwd, outputPathOptions.browserDirectory))],
+      runCommand: `node ${normalize(relative(cwd, outputBundleOptions.serverFilePath))}`,
+      neededDirs: [normalize(relative(cwd, outputBundleOptions.outputDirectory))],
+      staticAssets: [normalize(relative(cwd, outputBundleOptions.browserDirectory))],
     }),
   );
 }
 
 // Generate server file for CSR apps
-async function generateServer(outputPathOptions: OutputPathOptions): Promise<void> {
-  await mkdir(dirname(outputPathOptions.serverFilePath));
-  await copyFile(SIMPLE_SERVER_FILE_PATH, outputPathOptions.serverFilePath);
+async function generateServer(outputBundleOptions: OutputBundleOptions): Promise<void> {
+  await mkdir(dirname(outputBundleOptions.serverFilePath));
+  await copyFile(SIMPLE_SERVER_FILE_PATH, outputBundleOptions.serverFilePath);
+}
+
+// Validate output directory includes all necessary parts
+export async function validateOutputDirectory(
+  outputBundleOptions: OutputBundleOptions,
+): Promise<void> {
+  if (
+    !(await fsExtra.exists(outputBundleOptions.outputDirectory)) ||
+    !(await fsExtra.exists(outputBundleOptions.browserDirectory)) ||
+    !(await fsExtra.exists(outputBundleOptions.serverFilePath))
```

Comment:
Should we also verify the existence of bundleYamlPath

Reply:
Yes good call I added these checks